src/bundle/Resources/public/js/scripts/admin.location.view.js

```diff
@@ -59,6 +60,47 @@
         });
     };
     const generateLink = (locationId) => Routing.generate('_ezpublishLocation', { locationId });
+    const setModalTableTitle = (title) => {
+        const modalTableTitleNode = doc.querySelector(
+            `${SELECTOR_BULK_OPERATION_FAILED_MODAL} .ez-table-header .ez-table-header__headline`
+        );
+
+        modalTableTitleNode.innerHTML = title;
+    };
+    const setModalTableBody = (failedItemsData) => {
+        const modal = doc.querySelector(SELECTOR_BULK_OPERATION_FAILED_MODAL);
+        const table = modal.querySelector('.ez-bulk-operation-failed-modal__table');
+        const tableBody = table.querySelector('.ez-bulk-operation-failed-modal__table-body');
+        const tableRowTemplate = table.dataset.tableRowTemplate;
+        const fragment = doc.createDocumentFragment();
+
+        failedItemsData.forEach(({ contentName, contentTypeName }) => {
+            const container = doc.createElement('tbody');
+            const renderedItem = tableRowTemplate
+                .replace('{{ content_name }}', contentName)
+                .replace('{{ content_type_name }}', contentTypeName);
+
+            container.insertAdjacentHTML('beforeend', renderedItem);
+
+            const tableRowNode = container.querySelector('tr');
+
+            fragment.append(tableRowNode);
```

Comment:
You should not do it this way. It's not performant. The `append` should be invoked when all the actions, for each item, in a loop are done.

Reply:
You should care about it even if it's a `Fragment`;

Reply:
I did this as it was done here: https://github.com/ezsystems/ezplatform-page-builder/blob/911eb290f55a5c98638e5518cd35d340cb7764e5/src/bundle/Resources/public/js/config-form/widgets/collection.js#L59 

> The append should be invoked when all the actions, for each item, in a loop are done.
I'm calling `.append` at the end of each iteration. I'm not sure what changes you propose.